.github/workflows/labeler.yml

```diff
@@ -0,0 +1,27 @@
+﻿name: Auto Label
+
+on:
+  pull_request_target:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  labeler:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Run Auto Labeler
+      uses: actions/github-script@v7
+      with:
+        script: |
+          const { get_updated_label_set } = await import('${{ github.workspace }}/tools/pull_request_hooks/autoLabel.js');
+          const new_labels = await get_updated_label_set({ github, context });
+          github.rest.issues.setLabels({
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            labels: new_labels,
+          });
+          console.log(`Labels updated: ${new_labels}`);
+        github-token: ${{ steps.app-token-generation.outputs.token || secrets.GITHUB_TOKEN }}
```

Comment:
this seems massively unsafe, im 99% sure an attacker could open a pull request with a modified `autoLabel.js` script and github would run the script, no questions asked

Reply:
this was also done in `autoChangelog.js`, which also reeks, but it was safe by virtue of the auto changelog workflow only running when the PR was merged, meaning maints could sus out code injections before merging

this however, runs on pull request open.

Reply:
when I checked the docs it implied the default permissions meant you couldn't really do anything destructive (since it all defaults to read only (allegedly)), but double checking it and apparently, specifically for `pull_request_target`, the perms default to `read/write`. will address regardless

Reply:
also if it is an issue, seems like "remove guide comments" would have a similar issue as it runs on pr: opened

Reply:
FTR, it does not seem to be an issue. I tested it just to confirm my paranoia, and github seems to pull all workflow related from `master`, meaning you should not be able to inject a script that writes to the repo by just modifying it in a PR

Proof:
workflow run with a PR that injects a console.log in `autoLabel.js`, note how the log does not appear:
https://github.com/distributivgesetz/tgstation/actions/runs/12947217857/job/36113312890?pr=3

workflow run of another PR with the injection above merged, note how the log appears now:
https://github.com/distributivgesetz/tgstation/actions/runs/12947286029/job/36113528060?pr=4

Reply:
doesnt seem like your new code has any RCE exploits either, it should be fine